cos_registration_server/api/views.py

```diff
@@ -259,3 +266,110 @@ def delete(self, request: Request, uid: str) -> Response:
         dashboard = self._get_dashboard(uid)
         dashboard.delete()
         return Response(status=204)
+
+
+class PrometheusAlertRulesView(APIView):
+    """PrometheusAlertRules API view."""
+
+    def get(self, request: Request) -> Response:
+        """Prometheus Alert Rules get view.
+
+        request: Http GET request.
+        return: Http JSON response.
+        """
+        alert_rules = PrometheusAlertRule.objects.all()
+        rendered_rules = []
+
+        template_alert_rules = PrometheusAlertRule.objects.filter(
+            template=True
+        )
+        devices = Device.objects.all()
+
+        for device in devices:
+            for rule in device.prometheus_alert_rules.all():
+                if rule in template_alert_rules:
+                    env = Environment(
+                        variable_start_string="%%",
+                        variable_end_string="%%",
+                    )
+                    rule_string = yaml.dump(rule.rules)
+                    template = env.from_string(rule_string)
+                    context = {"juju_device_uuid": f"{device.uid}"}
+                    rendered_rule = template.render(context)
+                    rendered_rules.append(
+                        {
+                            "uid": rule.uid + "_" + device.uid,
+                            "rules": rendered_rule,
+                        }
+                    )
+        serialized = PrometheusAlertRuleSerializer(alert_rules, many=True)
+        serialized_list = list(serialized.data)
+        serialized_list.extend(rendered_rules)
+        return Response(serialized_list)
+
+    def post(self, request: Request) -> Response:
+        """Prometheus Alert Rules post view.
+
+        request: Http POST request.
+        return: Http JSON response.
+        """
+        serialized = PrometheusAlertRuleSerializer(data=request.data)
+        if serialized.is_valid():
+            serialized.save()
+            return Response(serialized.data, status=201)
+        return Response(serialized.errors, status=400)
+
+
+class PrometheusAlertRuleView(APIView):
+    """PrometheusAlertRule API view."""
+
+    def _get_alert_rule(self, uid: str) -> PrometheusAlertRule:
+        try:
+            alert_rule = PrometheusAlertRule.objects.get(uid=uid)
+            return alert_rule
+        except PrometheusAlertRule.DoesNotExist:
+            raise NotFound("Object does not exist")
+
+    def get(self, request: Request, uid: str) -> HttpResponse:
+        """Prometheus alert rule get view.
+
+        request: Http GET request.
+        return: Http JSON response.
+        """
+        alert_rule = self._get_alert_rule(uid)
+        serialized = PrometheusAlertRuleSerializer(alert_rule)
+
+        response = HttpResponse(
+            serialized.data["rules"],
+            content_type="application/yaml",
```

Comment:
We won't return the file here. We did that for dashboards because foxglove could load it on the fly.
Here we simply want to return the fields as json.

Reply:
removed file return, still returning the rule as yaml tho. So when someone gets /alert_rules/rule-uid it gets a vaild yaml

Reply:
We currently don't have a use case for that, while returning the dict of the object has a purpose: "patching and debugging"
Maybe later we could add an endpoint to get the file if we need that (something like `/api/v1/applications/loki/alert_rules/file`)